### ISSUE TITLE
Implementation of KL divergence in VAE example

### ISSUE THREAD

**Issue by jbschiratti** [pytorch/examples#824]
In this [VAE example](https://github.com/pytorch/examples/blob/master/vae/main.py), the loss function of the VAE is derived from equation (7) in [this paper](https://arxiv.org/pdf/1312.6114.pdf): `loss = -KLD + reconstruction`. In appendix B (top of page 11), the authors give a closed form expression for `-KLD`: `-KLD = 0.5 * sum(1 + logvar - mu.pow(2) - logvar.exp())`. Therefore, I would expect that: `loss = 0.5 * torch.sum(...) + BCE`. However, in the example, we have a negative sign in front of the KLD term (`loss = -0.5 * torch.sum(...) + BCE`). Is this a typo or did I miss something?

**Comment by soumith** [comment]
did you figure out the answer, or closed the issue for lack of activity?

**Comment by Hongru-Hu** [comment]
> did you figure out the answer, or closed the issue for lack of activity?

I think he realized that there's no problem with the KLD term

**Comment by jbschiratti** [comment]
Yes, the point is that one wants to _maximize_ (equiv. minimize the opposite of) the ELBO. Therefore the sign of the KLD term is correct.
